packages/runtimes/react-refresh/src/ReactRefreshRuntime.js

```diff
@@ -0,0 +1,40 @@
+// @flow strict-local
+
+import {Runtime} from '@parcel/plugin';
+import path from 'path';
+import invariant from 'assert';
+
+export default new Runtime({
+  apply({bundle, options}) {
+    if (bundle.type !== 'js' || !bundle.env.isBrowser() || !options.hot) {
+      return;
+    }
+
+    let mainEntry = bundle.getMainEntry();
+    let code;
+    if (mainEntry && mainEntry.meta.reactRefreshRuntimePath !== undefined) {
+      let runtime = mainEntry.meta.reactRefreshRuntimePath;
+      invariant(typeof runtime === 'string');
+
+      code = `
+var Refresh = require('${path.relative(__dirname, runtime)}');
+
+Refresh.injectIntoGlobalHook(window);
+window.$RefreshReg$ = function() {};
```

Comment:
Are we sure these functions aren't supposed to do something?

Reply:
Yes, every asset has a prelude, where there hooks are set to an actual implementation, and a postlude, where they are set to their previous values (like a stack). So the function at the very bottom has to be defined.
Metro does this and Dan also put this into his implementers guide.